crates/runtime/src/dataaccelerator/duckdb.rs

```diff
@@ -123,6 +126,19 @@ impl DataAccelerator for DuckDBAccelerator {
         vec!["db", "ddb", "duckdb"]
     }
 
+    fn file_path(&self, dataset: &Dataset) -> Option<String> {
+        self.duckdb_file_path(dataset)
+    }
+
+    fn is_initialized(&self, dataset: &Dataset) -> bool {
+        if !dataset.is_file_accelerated() {
+            return true; // memory mode DuckDB is always initialized
+        }
+
+        // otherwise, we're initialized if the file exists
+        self.has_existing_file(dataset)
```

Comment:
Is `has_existing_file` sufficient for judging whether accelerator is initialized? Both DuckDB and SQLite init has another step of initializing pool after validating the file path in `init` function, where accelerator could fail to be initialized as well.

Reply:
Initializing the pool is just a shortcut for creating the file. We do nothing with the pool, it gets dropped immediately. `init` also validates the file extension is correct, so if the file exists then we can assume initialization was successful.